clients/src/main/java/org/apache/kafka/common/record/CompressionType.java

```diff
@@ -156,7 +177,8 @@ else if (LZ4.name.equals(name))
     private static class LZ4Constructors {
         static final MethodHandle INPUT = findConstructor(
                 "org.apache.kafka.common.record.KafkaLZ4BlockInputStream",
```

Comment:
Yeah, but `KafkaLZ4BlockInputStream` itself should not be initialised until it's actually used. I made this change, used CompressionType.SNAPPY.wrap* methods while LZ4 was not in the classpath and it worked fine. The same exercise did not work for Snappy because `SnappyInputStream` was not in the classpath. Anyway, we can keep it as it is if you prefer as the performance impact is low when compared to other things (8ns instead of 4ns per construction).